pkg/apis/kops/cluster.go

```diff
@@ -650,3 +652,8 @@ type DNSControllerGossipConfig struct {
 	Secondary *DNSControllerGossipConfig `json:"secondary,omitempty"`
 	Seed      *string                    `json:"seed,omitempty"`
 }
+
+type RollingUpdate struct {
+	// TaintAllNeedUpdate taints all nodes in the instance group that need update before draining any of them
+	TaintAllNeedUpdate *bool `json:"taintAllNeedUpdate,omitempty"`
```

Comment:
We taint instead of cordon because cordoning can take the node out of rotation of the AWS ELB. Not a good thing to happen to all the nodes of an instancegroup.

I have a pending PR for allowing an external process to nominate nodes for updating, for example if they are older than site policy permits. Also, cluster autoscaler could add new nodes between spec update and rolling update. So interruption of rolling update isn't necessarily the only case where not all nodes in the instancegroup need updating.

It would be unfortunate if someone enabled this on an instancegroup that doesn't have autoscaling (or 100% surging), so I think having some indication to the admin that tainting will happen is desirable.